bin/server.js

```diff
@@ -1,16 +1,43 @@
 var mongoose    = require('mongoose');
 var express     = require('express');
+var bodyParser  = require('body-parser');
 var app         = express();
+var router      = express.Router();
+var routes      = require('../lib/routes');
 var mongoURI    = process.env.MONGO || 'mongodb://localhost:27017/nodeadvca';
 var log         = require('../middleware/logger').child({
   level : 'info',
   component : '[BIN-SERVER]'
 });
 
-// set port for server
-var port = process.env.PORT || 3000;
-
 // connect app to database
+log.info("Connecting to DB %s", mongoURI);
 mongoose.connect(mongoURI);
 
-app.listen(port);
+mongoose.connection.on('connected', function () {
+  log.info("Successfully connected to DB %s", mongoURI);
+  startServer();
+});
+mongoose.connection.on('error', function (err) {
+  log.error("Error connecting to DB %s", mongoURI, err);
+});
+
+// middleware to use for all requests
+router.use(function(req, res, next) {
```

Comment:
this is correct, however remember that we want to only allow the execution of server logic after the DB connection is successful and therefore this should sit within the startServer() method before you register the routes. Also, not a major and perhaps is due to code from different parts incl mine but you should get into practice of making consistant the use of either ' ' / " " although ideally ' ' is the standard